src/dawn/AST/ASTStmt.h

```diff
@@ -141,11 +185,23 @@ class BlockStmt : public Stmt {
 
   template <class Iterator>
   void insert_back(Iterator begin, Iterator end) {
+    std::for_each(begin, end, [&](const std::shared_ptr<Stmt>& stmt) {
+      DAWN_ASSERT(stmt);
```

Comment:
when you do a `for_each` anyway, why not just forward to `push_back`below?

Reply:
not sure, but maybe implementation of `std::vector` makes it more efficient to insert multiple items at once w.r.t. inserting them one by one...